radixdlt/src/main/java/org/radix/universe/system/LocalSystem.java

```diff
@@ -46,28 +50,58 @@
 public final class LocalSystem extends RadixSystem {
 	private final ECKeyPair keyPair;
 	private final Supplier<Map<String, Object>> counters;
+	private final InMemoryEpochInfo epochInfo;
 
 	LocalSystem() {
 		// Serializer only
-		this(ImmutableMap::of);
+		this(null, ImmutableMap::of);
 	}
 
 	@VisibleForTesting
-	LocalSystem(Supplier<Map<String, Object>> counters) {
+	LocalSystem(InMemoryEpochInfo epochInfo, Supplier<Map<String, Object>> counters) {
+		this.epochInfo = epochInfo;
 		this.keyPair = ECKeyPair.generateNew();
 		this.counters = counters;
 	}
 
-	public LocalSystem(Supplier<Map<String, Object>> counters, ECKeyPair key, String agent, int agentVersion, int protocolVersion, ImmutableList<TransportInfo> supportedTransports) {
+	public LocalSystem(
+		InMemoryEpochInfo epochInfo,
```

Comment:
I really don't like this interface, particularly the strong coupling between InMemoryEpochInfo and LocalSystem.

Can we abstract out the supply of the query details, perhaps through a generalisation of the approach for counters?